### PR TITLE
ESQL: Simple check if all blocks get released

### PR DESCRIPTION
**Pull request by alex-spies**
We need to avoid virtual memory leaks in ESQL's compute service, i.e. blocks not being released from the circuit breaker even though they are not referenced anymore.

Make the tests check for this for all operators whose tests inherit from `OperatorTestCase`.

### CODE REVIEW

**Comment by alex-spies** [comment]
Running
```bash
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.*.testSimpleDoesNotLeak"
```
Turns up virtual leaks in
- LimitOperator
- MvExpandOperator
- FilterOperator - but only sometimes! One can be reproduced with
```bash
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.FilterOperatorTests.testSimpleDoesNotLeak" -Dtests.seed=3F2BB02DF964C6F9 -Dtests.locale=pl-PL -Dtests.timezone=America/Kentucky/Louisville -Druntime.java=21
```

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java [hunk 1]
```diff
@@ -148,6 +149,47 @@ public final void testSimpleWithCranky() {
...
+        List<Page> result = drive(operator, input.iterator(), driverContext);
+
+        List<Block> resultBlocks = new ArrayList<>();
+        for (Page p : result) {
+            for (int i = 0; i < p.getBlockCount(); i++) {
+                resultBlocks.add(p.getBlock(i));
+            }
+        }
+
+        for (Page p : result) {
```
Comment (nik9000) [inline x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java:180]: I wonder if we can merge this into the `assertSimple`. I mean, if use the MockBlockFactory and check the breaker I think it'll already catch this as is. Those just aren't using that block factory.

**Comment by nik9000** [comment]
> LimitOperator

I'm looking at this one myself for some other tests. I'll see if I can make it happy.

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java (hunk 1 above)
Comment (ChrisHegarty) [reply x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java:180]: ++

Comment (alex-spies) [reply x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java]: Yep, works just fine. Done it this way now, this also increases the parametrization nicely as `assertSimple` is run for both small and large inputs.

**Comment by alex-spies** [comment]
> > LimitOperator
> 
> I'm looking at this one myself for some other tests. I'll see if I can make it happy.

Seems to have worked. Rebased on current main and don't see leaks in `Limit` anymore.

**Comment by alex-spies** [comment]
Disabled the check for the known offenders, `MvExpand` and `Filter.

We could merge this to avoid regressions in other operators, so I'm opening this for review.

**Comment by alex-spies** [comment]
Nope, @nik9000 , the limit operator is still leaking it seems. I got a case here:
```
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.LimitOperatorTests.testSimpleLargeInput" -Dtests.seed=6197048E15D119DB -Dtests.locale=ar-IQ -Dtests.timezone=Europe/Bratislava -Druntime.java=21
```

**Comment by alex-spies** [comment]
Muted the `LimitOperator` to make CI happy.

**Review by nik9000 — APPROVED** [review]
Lovely! Once you can make CI happy it'd be a lovely addition.

**Comment by nik9000** [comment]
> Nope, @nik9000 , the limit operator is still leaking it seems. I got a case here:

Fair enough!

**Pull request merged**